cosmo_tester/framework/examples/__init__.py

```diff
@@ -20,7 +20,182 @@
+                self.blueprint_file, self.blueprint_id)
+
+    def check_files(self, path=None, expected_content=None):
+        instances = self.manager.client.node_instances.list(
+            deployment_id=self.deployment_id,
+            _include=['id', 'node_id'],
+        )
+
+        if path is None:
+            path = self.inputs['path']
+        if expected_content is None:
+            expected_content = self.inputs['content']
+
+        for instance in instances:
+            if instance.node_id == 'file':
+                file_path = path + '_' + instance.id
+                content = self.example_host.get_remote_file_content(file_path)
+                assert content == expected_content
+
+    def uninstall(self, check_files_are_deleted=True):
+        self.logger.info('Cleaning up example.')
+        self.cfy.executions.start(
+            '--deployment-id', self.deployment_id,
+            '--tenant-name', self.tenant,
+            'uninstall'
+        )
+        self.installed = False
+        if check_files_are_deleted:
+            self.check_all_test_files_deleted()
+
+    def check_all_test_files_deleted(self, path=None):
+        if path is None:
+            path = self.inputs['path']
+
+        # This gets us the full paths, which then allows us to see if the test
+        # file prefix is in there.
+        # Technically this could collide if the string /tmp/test_file is in
+        # there but not actually part of the path, but that's unlikely so
+        # we'll tackle that problem when we cause it.
+        # Running with sudo to avoid exit status of 1 due to root owned files
+        tmp_contents = self.example_host.run_command('sudo find /tmp').stdout
+        assert path not in tmp_contents
+
+    def upload_and_verify_install(self, timeout=900):
+        self.upload_blueprint()
+        self.create_deployment()
+        self.install(timeout)
+        self.assert_deployment_events_exist()
+        self.check_files()
+
+    def create_deployment(self):
+        self.logger.info(
+                'Creating deployment [id=%s] with the following inputs:\n%s',
+                self.deployment_id,
+                json.dumps(self.inputs, indent=2))
+        with set_client_tenant(self.manager, self.tenant):
+            self.manager.client.deployments.create(
+                deployment_id=self.deployment_id,
+                blueprint_id=self.blueprint_id,
+                inputs=self.inputs)
+        self.cfy.deployments.list(tenant_name=self.tenant)
+
+    def install(self, timeout=900):
+        self.logger.info('Installing deployment...')
+        try:
+            self.cfy.executions.start.install(['-d', self.deployment_id,
+                                               '-t', self.tenant,
+                                               '--timeout', timeout])
+            self.installed = True
+        except Exception as e:
+            if 'if there is a running system-wide' in e.message:
```

Comment:
plz no e.message

Reply:
Ah, sorry- I copied that one from the Abstract one. Will change it.